### PR TITLE
NEXUS-20443 Add integration tests for R Proxy

### PR DESCRIPTION
**Pull request by aornatovskyy**
https://issues.sonatype.org/browse/NEXUS-20443

### CODE REVIEW

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class RITSupport
+    extends RepositoryITSupport
+{
+  public static final String R = "r";
+
+  public static final String PATH = "bin/macosx/el-capitan/contrib/3.6";
+
+  public static final String AGRICOLAE = "agricolae";
```
Comment (sergeyonofreychuck) [inline nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java]: improve naming

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  public static final String AGRICOLAE_VERSION = "1.3-1";
+
+  public static final String TGZ = ".tgz";
+
+  public static final String GZ = ".gz";
+
+  public static final String AGRICOLAE_FILE_NAME = String.format("%s_%s%s", AGRICOLAE, AGRICOLAE_VERSION, TGZ);
```
Comment (sergeyonofreychuck) [inline nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java]: AGRICOLAE_PACKAGE_FILE_NAME

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import org.sonatype.nexus.testsuite.testsupport.RepositoryITSupport;
+
+import org.junit.Rule;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class RITSupport
+    extends RepositoryITSupport
+{
+  public static final String R = "r";
```
Comment (sergeyonofreychuck) [inline nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java]: R_FORMAT_NAME ?

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  public static final String PATH = "bin/macosx/el-capitan/contrib/3.6";
```
Comment (sergeyonofreychuck) [inline nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java]: what path?

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  public static final String AGRICOLAE = "agricolae";
+
+  public static final String AGRICOLAE_VERSION = "1.3-1";
+
+  public static final String TGZ = ".tgz";
```
Comment (sergeyonofreychuck) [inline nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java]: TGZ_EXT ?

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  public static final String CONTENT_TYPE_GZIP = "application/x-gzip";
+
+  public static final String PACKAGES = "PACKAGES";
+
+  public static final String PACKAGES_FILE_NAME = String.format("%s%s", PACKAGES, GZ);
+
+  public static final String PACKAGES_FULL = String.format("%s/%s", PATH, PACKAGES_FILE_NAME);
+
+  @Rule
+  public RepositoryRuleR repos = new RepositoryRuleR(() -> repositoryManager);
```
Comment (sergeyonofreychuck) [inline nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java]: why not protected?

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java (hunk 1 above)
Comment (aornatovskyy) [reply nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java]: d1f32cb0edf114f5b6aeba9f35cb221897f29061

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java (hunk 2 above)
Comment (aornatovskyy) [reply nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java]: d1f32cb0edf114f5b6aeba9f35cb221897f29061

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java (hunk 3 above)
Comment (aornatovskyy) [reply nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java]: d1f32cb0edf114f5b6aeba9f35cb221897f29061

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java (hunk 5 above)
Comment (aornatovskyy) [reply nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java]: d1f32cb0edf114f5b6aeba9f35cb221897f29061

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java (hunk 6 above)
Comment (aornatovskyy) [reply nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RITSupport.java]: d1f32cb0edf114f5b6aeba9f35cb221897f29061

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/fixtures/RepositoryRuleR.groovy [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.plugins.r.internal.fixtures;
+
+import javax.inject.Provider;
+
+import org.sonatype.nexus.repository.manager.RepositoryManager;
+import org.sonatype.nexus.testsuite.testsupport.fixtures.RepositoryRule;
+
+class RepositoryRuleR
```
[the remark this answers is not in the document]
Comment (aornatovskyy) [reply nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/fixtures/RepositoryRuleR.groovy]: Took this approach from Mark example for conda. But in any case, I can't use only `RepositoryRuleR`. Just tried, and it doesn't work.
In the same time `RepositoryRuleR`   implements `RRepoRecipes` which has a default implementation for creating a repository.

File: nexus-repository-r-it/pom.xml [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+    <!--
+    Pax-Exam test driver - end
+    -->
+
+    <!--
+    only needed due to testsupport reference to: org.sonatype.nexus.repository.maven.VersionPolicy
+    -->
+    <dependency>
+      <groupId>org.sonatype.nexus.plugins</groupId>
+      <artifactId>nexus-repository-maven</artifactId>
```
Comment (doddi) [inline nexus-repository-r-it/pom.xml]: Suggestion: Maybe raise an improvement ticket to remove the need for this dependancy

File: nexus-repository-r/pom.xml [hunk 9]
```diff
@@ -26,6 +26,50 @@
...
+      <version>1.4.1</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.sonatype.goodies</groupId>
+      <artifactId>goodies-testsupport</artifactId>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
```
Comment (doddi) [inline nexus-repository-r/pom.xml]: Question: Is this needed?

File: nexus-repository-r/pom.xml [hunk 10]
```diff
@@ -26,6 +26,50 @@
...
+      <artifactId>nexus-repository</artifactId>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-compress</artifactId>
+    </dependency>
+
+    <dependency>
```
Comment (doddi) [inline nexus-repository-r/pom.xml]: Question: Is this needed?

File: nexus-repository-r-it/pom.xml (hunk 8 above)
Comment (aornatovskyy) [reply nexus-repository-r-it/pom.xml]: [NEXUS-20960](https://issues.sonatype.org/browse/NEXUS-20960)

File: nexus-repository-r/pom.xml (hunk 10 above)
Comment (aornatovskyy) [reply nexus-repository-r/pom.xml]: It is used in the `RPackageVersion`

File: nexus-repository-r/pom.xml (hunk 9 above)
Comment (aornatovskyy) [reply nexus-repository-r/pom.xml]: it is used in the `r` module unit tests.

**Pull request merged**